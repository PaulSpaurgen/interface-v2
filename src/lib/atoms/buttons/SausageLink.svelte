<script>
	import { staticImages } from '$lib/components/images/staticImages';

	export let href = '/';
	export let text = 'Get your on-chain domain with 3DNS';
</script>

<a
	class="flex w-fit items-center gap-2 rounded-lg bg-white px-4 py-3 hover:bg-white/75"
	{href}
	target="_blank"
>
	<div class="mr-1 flex flex-col items-start text-sm font-medium">
		<span class="">{text}</span>
	</div>
	<img class="my-auto" src={staticImages.OpenInNew} alt="Open" />
</a>
