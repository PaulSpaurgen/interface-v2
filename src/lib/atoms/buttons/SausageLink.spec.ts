import { describe, it, afterEach } from 'vitest';
import { cleanup, render } from '@testing-library/svelte';
import SausageLink from './SausageLink.svelte';

describe('SausageLink', () => {
	afterEach(() => {
		cleanup();
	});

	it('renders without errors', () => {
		const { container } = render(SausageLink);
		expect(container).toMatchSnapshot();
	});

	it('renders an anchor tag with passed in href', () => {
		const { getByRole } = render(SausageLink, { props: { href: 'thisisatesthrefforsausagelink' } });
		const anchor = getByRole('link');
		expect(anchor.getAttribute('href')).toBe('thisisatesthrefforsausagelink');
	});

	it('renders the text passed to it', () => {
		const { getByText } = render(SausageLink, {
			props: { text: 'thisisthecustomtextforsausagebutton' }
		});
		expect(getByText('thisisthecustomtextforsausagebutton')).toBeTruthy();
	});

	it('renders the image which indicates it is a link', () => {
		const { getByAltText } = render(SausageLink, {
			props: {
				href: 'the link href',
				text: 'the link text'
			}
		});

		const image = getByAltText('Open');
		const staticImagesImported = { OpenInNew: '/images/openinnew.svg' };
		expect(image.getAttribute('src')).toBe(staticImagesImported.OpenInNew);
	});
});
