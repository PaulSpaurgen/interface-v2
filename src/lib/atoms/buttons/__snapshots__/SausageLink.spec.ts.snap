// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SausageLink > renders without errors 1`] = `
<body>
  <div>
    <a
      class="flex w-fit items-center gap-2 rounded-lg bg-white px-4 py-3 hover:bg-white/75"
      href="/"
      target="_blank"
    >
      <div
        class="mr-1 flex flex-col items-start text-sm font-medium"
      >
        This is text for sausage link ğŸŒ­
      </div>
       
      <img
        alt="Open"
        class="my-auto"
        src="/images/openinnew.svg"
      />
    </a>
    <!--&lt;SausageLink&gt;-->
  </div>
</body>
`;
